modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/GridDhtPartitionTopologyImpl.java

```diff
@@ -2702,20 +2684,27 @@ private void removeNode(UUID nodeId) {
         try {
             AffinityTopologyVersion lastAffChangeVer = ctx.exchange().lastAffinityChangedTopologyVersion(lastTopChangeVer);
 
-            if (lastAffChangeVer.compareTo(rebFinishedTopVer) > 0 && log.isInfoEnabled())
-                log.info("Affinity topology changed, no MOVING partitions will be owned " +
-                    "[rebFinishedTopVer=" + rebFinishedTopVer +
-                    ", lastAffChangeVer=" + lastAffChangeVer + "]");
+            if (lastAffChangeVer.compareTo(rebFinishedTopVer) > 0) {
+                if (log.isInfoEnabled()) {
+                    log.info("Affinity topology changed, no MOVING partitions will be owned " +
+                        "[rebFinishedTopVer=" + rebFinishedTopVer +
+                        ", lastAffChangeVer=" + lastAffChangeVer + "]");
+                }
+
+                grp.preloader().forceRebalance();
```

Comment:
Do we have a test for this particular scenario ?

Reply:
That case fires on test LocalWalModeChangeDuringRebalancingSelfTest#testWithExchangesMerge